### DISCUSSION TITLE
cl65 doesn't know what to do when passed a line ending in carriage return+line feed

### DISCUSSION THREAD

**Discussion by jeffythedragonslayer** [cc65/cc65#1803] [Q&A]
Within cygwin I tried to run a `build.sh` script containing this:

```
#!/bin/bash
cl65 -C smc.cfg -o hello.smc -l hello.list -m hello.mmap hello.asm

```

It ends with a carriage return and line feed, because when I saved it in vi it had completed the incomplete line ending.  That caused cl65 to get confused and emit this error:

`'l65.exe: Don't know what to do with 'hello.asm`

Workaround: open the file in notepad and backspace out the carriage return/line feed

**Comment by spiro-trikaliotis** [comment]
If you install cygwin in "LF mode", then this is what happens.
If you install cygwin in "CR/LF mode", then it handles this correctly.

This is a problem of cygwin, not of cc65/cl65. Other programs, even cygwin's own ones, show the same behaviour.

At least, this all was the case when I last used cygwin years ago.

**Comment by jeffythedragonslayer** [comment]
oh, do you know if cygwin supports changing the mode without reinstalling it?  I don't see "LF mode" or "CR/LF mode" in the user's guide PDF

**Comment by spiro-trikaliotis** [comment]
Ok, I see that cygwin does not support this option anymore:
https://superuser.com/questions/1186813/cygwin-command-line-installer-option-for-unix-dos-line-endings

However, it is still a problem of cygwin, which gives a wrong command-line to cl65, not of cl65.

You could use the tool dos2unix to oonvert the file from cr/lf line endings to lf line endings instead of using notepad to remove the last cr/lf (which, by Unix definition, results in a non-text file, because the last lf is missing...)

**Comment by mrdudz** [comment]
moved to discussion, this isnt really a bug

**Comment by groessler** [comment]
What needs to be discussed here? It's obviously wrong to use a shell script with CR/LF line endings.
Interesting is that /bin/bash is executed. I know the effect that the system doesn't find /bin/bash\r
